e2e/pos_test.go

```diff
@@ -82,6 +83,119 @@ func validateValidatorSet(
+	numNewStakers := 2
+
+	for i := 0; i < numNewStakers; i++ {
+		k, a := tests.GenerateKeyAndAddr(t)
+
+		accounts = append(accounts, struct {
+			key     *ecdsa.PrivateKey
+			address types.Address
+		}{
+			key:     k,
+			address: a,
+		})
+	}
+
+	defaultBalance := framework.EthToWei(100)
+	ibftManager := framework.NewIBFTServersManager(
+		t,
+		numGenesisValidators,
+		IBFTDirPrefix,
+		func(i int, config *framework.TestServerConfig) {
+			config.SetSeal(true)
+			config.SetEpochSize(2)
+			config.PremineValidatorBalance(defaultBalance)
+			for j := 0; j < numNewStakers; j++ {
+				config.Premine(accounts[j].address, defaultBalance)
+			}
+			config.SetIBFTPoS(true)
+			config.SetMinValidatorCount(minValidatorCount)
+			config.SetMaxValidatorCount(maxValidatorCount)
+		})
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	ibftManager.StartServers(ctx)
+
+	srv := ibftManager.GetServer(0)
+
+	client := srv.JSONRPC()
+
+	testCases := []struct {
+		name              string
+		address           types.Address
+		key               *ecdsa.PrivateKey
+		expectedExistence bool
+		expectedSize      int
+	}{
+		{
+			name:              "Can add a 5th validator",
+			address:           accounts[0].address,
+			key:               accounts[0].key,
+			expectedExistence: true,
+			expectedSize:      numGenesisValidators + 1,
+		},
+		{
+			name:              "Can not add a 6th validator",
+			address:           accounts[1].address,
+			key:               accounts[1].key,
+			expectedExistence: false,
+			expectedSize:      numGenesisValidators + 1,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			err := framework.StakeAmount(tt.address, tt.key, stakeAmount, srv)
+			assert.NoError(t, err)
+			validateValidatorSet(t, tt.address, client, tt.expectedExistence, tt.expectedSize)
+		})
+	}
+}
+
+func TestPoS_DefaultValidatorBoundaries(t *testing.T) {
+	// Test scenario -> There are 4 validators, and the validator limit is default (1, maxUint32)
+	// When trying to add a new validator, it should be added.
+	stakerKey, stakerAddr := tests.GenerateKeyAndAddr(t)
+	stakeAmount := framework.EthToWei(1)
+	numGenesisValidators := IBFTMinNodes
+
+	defaultBalance := framework.EthToWei(100)
+	ibftManager := framework.NewIBFTServersManager(
+		t,
+		numGenesisValidators,
+		IBFTDirPrefix,
+		func(i int, config *framework.TestServerConfig) {
+			config.SetSeal(true)
+			config.SetEpochSize(2)
+			config.PremineValidatorBalance(defaultBalance)
+			config.Premine(stakerAddr, defaultBalance)
+			config.SetIBFTPoS(true)
+		})
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	ibftManager.StartServers(ctx)
+
+	srv := ibftManager.GetServer(0)
+
+	client := srv.JSONRPC()
+
+	err := framework.StakeAmount(stakerAddr, stakerKey, stakeAmount, srv)
+	assert.NoError(t, err)
+	validateValidatorSet(t, stakerAddr, client, true, numGenesisValidators+1)
+}
```

Comment:
What does this test actually test, that `TestPoS_ValidatorBoundaries` doesn't already cover?

Reply:
fixed a91c866e0f6d742a0cc4af63f07196b4703ec96a

You are right, this one should be deleted